packages/block-library/src/query-title/index.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Server-side rendering of the `core/query-title` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/query-title` block on the server.
+ * For now it only supports Archive title,
+ * using queried object information
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the query title based on the queried object.
+ */
+function render_block_core_query_title( $attributes, $content, $block ) {
+	$type       = isset( $attributes['type'] ) ? $attributes['type'] : null;
+	$is_archive = is_archive();
+	if ( ! $type || ( 'archive' === $type && ! $is_archive ) ) {
+		return '';
+	}
+	$title = '';
+	if ( $is_archive ) {
+		$title = get_the_archive_title();
+	}
+	$tag_name           = isset( $attributes['level'] ) ? 'h' . (int) $attributes['level'] : 'h2';
```

Comment:
Should this default to `h2` or `h1`? 🤔 
Thining of where this will most likely be used, it will be as the main title on archives, correct? In that case maybe `h1` would be a saner default

Reply:
In general I agree with you, but I'm afraid of doing it as there might be some themes (like tt1-blocks) that use the `gigantic` font size for h1. I think it's a small change that we can revisit if we need to. What do you think?

Reply:
Semantically it should probably be `h1`... If tt1-blocks uses a gigantic font-size then that's a theme issue and should be fixed there 😄